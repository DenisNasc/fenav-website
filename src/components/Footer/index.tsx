import React from 'react';
import {AppBar, Container, Typography, Link, Divider} from '@material-ui/core';
import {createStyles, Theme, makeStyles} from '@material-ui/core/styles';

const Footer = () => {
  const classes = useStyles();
  return (
    <AppBar
      color="secondary"
      elevation={0}
      component="footer"
      classes={{root: classes.footer}}
      position="relative"
    >
      <Container className={classes.sitesRelacionadosContainer}>
        <Typography component="div" className={classes.title}>
          Sites Relacionados
        </Typography>
        <Container className={classes.sitesRelacionados}>
          <Typography component="div" className={classes.textos}>
            <Link href="https://portal.ufpa.br/">Universidade Federal do Pará (UFPA)</Link>
          </Typography>
          <Typography component="div" className={classes.textos}>
            <Link href="http://www.itec.ufpa.br/">Instituto de Tecnologia da UFPA (ITEC)</Link>
          </Typography>
          <Typography component="div" className={classes.textos}>
            <Link href="https://www.ana.gov.br/">Agência Nacional de Águas (ANA)</Link>
          </Typography>
          <Typography component="div" className={classes.textos}>
            <Link href="http://portal.antaq.gov.br/">
              Agencia Nacional de Transporte Aquaviário (ANTAQ)
            </Link>
          </Typography>
          <Typography component="div" className={classes.textos}>
            <Link href="https://www.marinha.mil.br/">Marinha do Brasil</Link>
          </Typography>
          <Typography component="div" className={classes.textos}>
            <Link href="https://www.marinha.mil.br/cpaor/">
              Capitania dos Portos da Amazônia Oriental
            </Link>
          </Typography>
        </Container>
      </Container>

      <Divider orientation="vertical" flexItem variant="middle" />

      <Container className={classes.contatosFaculdadeContainer}>
        <Typography component="div" className={classes.title}>
          Contatos
        </Typography>
        <Container className={classes.contatosFaculdade}>
          <Typography component="div" className={classes.textos}>
            Secretaria de Graduação: (91)998247565 (Rubenson)
          </Typography>
          <Typography component="div" className={classes.textos}>
            Secretaria de Pós-Graduação: (91)998247565 (Nayrama)
          </Typography>
          <Typography className={classes.textos}>
            Centro Acadêmico: (91)998247565 (Isabelly)
          </Typography>
        </Container>
      </Container>

      <Divider orientation="vertical" flexItem variant="middle" />

      <Container className={classes.enderecoContainer}>
        <Typography className={classes.title}>Endereço</Typography>
        <Container className={classes.endereco}>
          <Typography className={classes.textos}>
            Condomínio Fernando Guilhon BL. 10B APTO. 304
          </Typography>
        </Container>
      </Container>
    </AppBar>
  );
};

export default Footer;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      top: 'auto',
      bottom: 0,
      height: '100%',
      paddingLeft: '10px',
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'flex-start',
      borderTop: '1px solid black',
    },
    textos: {
      fontSize: '12px',
    },
    title: {
      fontSize: '16px',
    },
    sitesRelacionados: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'column',
      alignItems: 'flex-start',
      justifyContent: 'center',
      height: '50px',
      width: '100%',
      margin: '0px',
      padding: '0px',
    },
    sitesRelacionadosContainer: {
      display: 'flex',
      flexDirection: 'column',
      height: '100%',
      width: '800px',
      margin: '0px',
      padding: '0px',
    },

    contatosFaculdade: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'column',
      alignItems: 'flex-start',
      justifyContent: 'center',
      height: '50px',
      width: '100%',
      margin: '0px',
      padding: '0px',
    },
    contatosFaculdadeContainer: {
      display: 'flex',
      flexDirection: 'column',
      height: '100%',
      width: '650px',
      margin: '0px',
      padding: '0px',
    },

    endereco: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'column',
      alignItems: 'flex-start',
      justifyContent: 'center',
      height: '50px',
      width: '100%',
      margin: '0px',
      padding: '0px',
    },
    enderecoContainer: {
      display: 'flex',
      flexDirection: 'column',
      height: '100%',
      width: '300px',
      margin: '0px',
      padding: '0px',
    },
  })
);
