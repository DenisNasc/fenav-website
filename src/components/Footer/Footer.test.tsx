import React from 'react';
import ReactDOM from 'react-dom';
import {act} from 'react-dom/test-utils';
import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '.';

describe('Footer Component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Footer />, div);
  });

  test('should match to the snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
