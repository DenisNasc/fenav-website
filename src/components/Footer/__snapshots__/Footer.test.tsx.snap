// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match to the snapshot 1`] = `
<footer
  className="MuiPaper-root MuiAppBar-root makeStyles-footer-1 MuiAppBar-positionRelative MuiAppBar-colorSecondary MuiPaper-elevation0"
>
  <div
    className="MuiContainer-root makeStyles-sitesRelacionadosContainer-5 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
    >
      Sites Relacionados
    </div>
    <div
      className="MuiContainer-root makeStyles-sitesRelacionados-4 MuiContainer-maxWidthLg"
    >
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://portal.ufpa.br/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Universidade Federal do Pará (UFPA)
        </a>
      </div>
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="http://www.itec.ufpa.br/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Instituto de Tecnologia da UFPA (ITEC)
        </a>
      </div>
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://www.ana.gov.br/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Agência Nacional de Águas (ANA)
        </a>
      </div>
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="http://portal.antaq.gov.br/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Agencia Nacional de Transporte Aquaviário (ANTAQ)
        </a>
      </div>
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://www.marinha.mil.br/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Marinha do Brasil
        </a>
      </div>
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://www.marinha.mil.br/cpaor/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Capitania dos Portos da Amazônia Oriental
        </a>
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-flexItem MuiDivider-vertical"
  />
  <div
    className="MuiContainer-root makeStyles-contatosFaculdadeContainer-7 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
    >
      Contatos
    </div>
    <div
      className="MuiContainer-root makeStyles-contatosFaculdade-6 MuiContainer-maxWidthLg"
    >
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        Secretaria de Graduação: (91)998247565 (Rubenson)
      </div>
      <div
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        Secretaria de Pós-Graduação: (91)998247565 (Nayrama)
      </div>
      <p
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        Centro Acadêmico: (91)998247565 (Isabelly)
      </p>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-flexItem MuiDivider-vertical"
  />
  <div
    className="MuiContainer-root makeStyles-enderecoContainer-9 MuiContainer-maxWidthLg"
  >
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
    >
      Endereço
    </p>
    <div
      className="MuiContainer-root makeStyles-endereco-8 MuiContainer-maxWidthLg"
    >
      <p
        className="MuiTypography-root makeStyles-textos-2 MuiTypography-body1"
      >
        Condomínio Fernando Guilhon BL. 10B APTO. 304
      </p>
    </div>
  </div>
</footer>
`;
